src/Symfony/Bundle/FrameworkBundle/Console/Application.php

```diff
@@ -70,12 +73,28 @@ public function doRun(InputInterface $input, OutputInterface $output)
 
         $this->setDispatcher($this->kernel->getContainer()->get('event_dispatcher'));
 
+        if ($this->registrationErrors) {
+            $this->renderRegistrationErrors($output);
```

Comment:
actually it's triggered at application construction https://github.com/symfony/symfony/blob/3.4/src/Symfony/Component/Console/Application.php#L92 :) (calls `add()` which then calls `registerCommands()`)